pkg/proxy/types.go

```diff
@@ -48,8 +49,14 @@ func (spn ServicePortName) String() string {
 type ServicePort interface {
 	// String returns service string.  An example format can be: `IP:Port/Protocol`.
 	String() string
-	// ClusterIPString returns service cluster IP in string format.
-	ClusterIPString() string
+	// GetClusterIP returns service cluster IP in net.IP format.
+	GetClusterIP() net.IP
```

Comment:
golang style is not to use "Get" in getter names (https://golang.org/doc/effective_go.html#Getters)

Reply:
I used "Get" in the getter name because some of them already did and now a majority of them do now. Should I remove "Get" from all of them?

I changed from string to net.IP because sometimes the string version of the IP address and other times they just accessed the net.IP through BaseServiceInfo.ClusterIP. I thought instead of having two getters since net.IP has a string method it would be better to return the more generic type and call the string function when needed

Reply:
Hm... it appears that the interface used to have `ClusterIP()`, `Protocol()`, and `HealthCheckNodePort()`, and @MrHohn changed it in https://github.com/kubernetes/kubernetes/commit/b485f7b5 as part of #56880. (Renaming the methods to have `Get` so as to not conflict with newly-added public struct fields named `ClusterIP`, `Protocol`, and `HealthCheckNodePort`.) I'm not sure if there's any good reason for it...

Anyway, if you're making the struct fields private and consistently using the interface methods, then you should make the interface method names not have "Get" in them.

Reply:
Thanks for adding me. @danwinship is correct, I made these this way purely to avoid conflicting. +1 to not having the redundant `Get` if no longer needed.